### DISCUSSION TITLE
[Zstd] About recommended usage

### DISCUSSION THREAD

**Discussion by hyunsooda** [klauspost/compress#1030] [Q&A]
According to the Zstd [description](https://github.com/klauspost/compress/tree/master/zstd#zstd),  There are two methods provided: stream and block.
Is there a recommended selection based on the data size to be compressed? (e.g., for the size 100MB, which one is better).
In the first statement in `Blocks` section, 
> For compressing small blocks,

Providing a more detailed description or an approximate number would be helpful, especially for newcomers.
Thank you.

**Comment by klauspost** [comment]
Blocks are only really used as an internal structure. I have never seen them used standalone. And they are limited to 128KB.

This library does not expose any block encoder/decoder.

**Comment by hyunsooda** [comment]
Hi @klauspost, I meant the stream method to [this](https://github.com/klauspost/compress/tree/master/zstd#usage) and block to [this](https://github.com/klauspost/compress/tree/master/zstd#blocks) (`EncodeAll()`).

In my local experiment for multiple 100MB data, the two methods do not seem very different. So, which method is preferable based on the origin target size to be compressed?

**Reply by klauspost** [reply]
One is streaming, another is not. The streaming will use multiple goroutines, the single-block will not.

**Reply by hyunsooda** [reply]
Yes, in my case the multiple of 100MB data is independent, so it can be compressed independently. So, my original question was, which is better based on the target size to be compressed..? For example, I expected that streaming is preferable when the data size range is within A to B, otherwise `EncodeAll` is prefereable.

**Reply by klauspost** [reply]
The output is very close to, if not the exact same.

**Reply by hyunsooda** [reply]
Got it. Thank you. Do you mean the output contains not only the compression rate but also the compression time?

**Reply by klauspost** [reply]
Streams will typically encode about twice as fast, and should generally be used for anything above 1MB or so.

**Reply by hyunsooda** [reply]
Thanks for the valuable information. Let me close.

**Comment by hyunsooda** [comment]
-
